src/app/core/public/api/impl/Circuit.ts

```diff
@@ -155,7 +166,34 @@ export abstract class CircuitImpl implements Circuit {
         //  For now, ignore the `space`, and ignore any non-Component
         //   objects that are selected
         //  From these components, average their positions
-        throw new Error("Method not implemented.");
+
+        const allComponents = this.selections.get();
+
+        let xPosition = 0;
+        let yPosition = 0;
+        
+        // Case: only one component selected
+        if (allComponents.length === 1) {
+            const comp = this.getComponent(allComponents[0]);
+            if (comp !== undefined)
+                return comp.pos;
+        }
+
+        // Case: multiple components are selected
+        for (let id of allComponents) {
+            const comp = this.getComponent(id);
+            if (comp !== undefined) {
+                xPosition += comp.pos.x;
+                yPosition += comp.pos.y;
+            }
+        }
+
+        // Calculate average position
+        xPosition = xPosition / allComponents.length;
+        yPosition = yPosition / allComponents.length;
```

Comment:
I'm going to have you attempt to condense this using `.map` and `.filter` as I showed above! \
(hint: take advantage of `comp.pos` being a `Vector` and don't use x/y separately)\
(hint 2: look into the `.reduce` function, which can take an array of things (like numbers/vector), and reduce them into a single number/vector)